File: core/src/com/broll/gainea/server/core/cards/impl/direct/C_War.java
package com.broll.gainea.server.core.cards.impl.direct;

import com.broll.gainea.server.core.cards.DirectlyPlayedCard;
import com.broll.gainea.server.core.map.Location;
import com.broll.gainea.server.core.objects.BattleObject;
import com.broll.gainea.server.core.player.Player;
import com.broll.gainea.server.core.utils.PlayerUtils;
import com.broll.gainea.server.core.utils.StreamUtils;
import com.broll.gainea.server.core.utils.UnitControl;

import java.util.ArrayList;
import java.util.List;

public class C_War extends DirectlyPlayedCard {

    public C_War() {
        super(9, "Wilde Schlacht", "Jeder Spieler wählt ein feindlich besetztes Land. Jeder Einheit darauf wird 1 Schaden zugefügt.");
        setDrawChance(0.5f);
    }

    @Override
    protected void play() {
        PlayerUtils.iteratePlayers(game, 500, player -> {
            List<Location> locations = new ArrayList<>(PlayerUtils.getHostileLocations(game, player));
            if (!locations.isEmpty()) {
                Location location = selectHandler.selectLocation("Wähle feindliches Land", locations);
                StreamUtils.safeForEach(location.getInhabitants().stream().filter(it -> it instanceof BattleObject).map(it -> (BattleObject) it),
                        unit -> UnitControl.damage(game, unit, 1));
            }
        });
    }

}

File: server/src/main/com/broll/gainea/server/core/cards/impl/direct/C_War.kt
package com.broll.gainea.server.core.cards.impl.direct

import com.broll.gainea.server.core.cards.DirectlyPlayedCard
import com.broll.gainea.server.core.cards.EffectType.DISRUPTION
import com.broll.gainea.server.core.utils.UnitControl.damage
import com.broll.gainea.server.core.utils.getEnemyLocations
import com.broll.gainea.server.core.utils.iteratePlayers

class C_War : DirectlyPlayedCard(
    9,
    DISRUPTION,
    "Wilde Schlacht",
    "Jeder Spieler wählt ein feindlich besetztes Land. Jeder Einheit darauf wird 1 Schaden zugefügt."
) {
    init {
        drawChance = 0.5f
    }

    override fun play() {
        game.iteratePlayers(500) { player ->
            val locations = game.getEnemyLocations(player).toList()
            if (locations.isNotEmpty()) {
                val location =
                    selectHandler.selectLocation(player, "Wähle feindliches Land", locations)
                location.units.forEach { game.damage(it) }
            }
        }
    }
}
